lib/beaker/dsl/structure.rb

```diff
@@ -213,7 +213,6 @@ def confine(type, criteria, host_array = nil, &block)
           logger.warn "No suitable hosts with: #{criteria.inspect}"
           skip_test 'No suitable hosts found'
         end
-        self.hosts = hosts_to_modify
```

Comment:
My understanding is that this modifies the global hosts array; We shouldn't do that.